app/models/assignment_repo/creator.rb

```diff
@@ -189,5 +190,63 @@ def suffixed_repo_name(repository_name, suffix_count)
       suffix = "-#{suffix_count}"
       repository_name.truncate(100 - suffix.length, omission: '') + suffix
     end
+
+    # Handles assignment repository setup
+    #
+    # configs_repo    - GitHubRepository containing the configuration files
+    # assignment_repo - GitHubRepository for which to perform the configuration
+    #                   setups
+    #
+    # Returns nothing
+    def setup_repository(configs_repo, assignment_repo)
+      return unless configs_repo.classroom_configs?
+      conf_tree = configs_repo.branch('github-classroom').commit.commit.tree.sha
+      configs_repo.tree(conf_tree).tree.each do |conf|
+        process_config(configs_repo, assignment_repo, conf)
+      end
+
+      wait_import_completion assignment_repo
+
+      assignment_repo.remove_branch('github-classroom')
+    end
+
+    def wait_import_completion(github_repository)
+      t = 1
+      loop do
+        break if github_repository.import_progress[:status] == 'complete' || t > 60
```

Comment:
`|| t > 60` makes it seem like this times out in 60 seconds, but that isn't actually what's happening here. If we do end up going with this sleep logic, can we make the timeout more clear?